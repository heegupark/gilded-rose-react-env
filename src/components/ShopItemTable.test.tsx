import React from 'react';
import { shallow } from 'enzyme'
import ShopItemTable from "./ShopItemTable";
import { items } from '../data/shopItems';

describe('ShopItemTable', () => {
    it('should render <ShopItemTable />', () => {
        const component = shallow(<ShopItemTable items={items} />);
        expect(component.debug()).toMatchSnapshot();
        expect(component.find('Name')).toBeDefined();
        expect(component.find('Quality')).toBeDefined();
        expect(component.find('Sell In Days')).toBeDefined();
    });

    it('renders learn react link', () => {
        const component = shallow(<ShopItemTable items={items} />);
        const header = component.find('.item-row');
        expect(header.length).toBe(5);
    });
});
