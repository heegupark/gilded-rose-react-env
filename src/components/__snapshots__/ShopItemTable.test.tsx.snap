// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopItemTable should render <ShopItemTable /> 1`] = `
"<ForwardRef striped={true} bordered={true} hover={true}>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Quality
      </th>
      <th>
        Sell In Days
      </th>
    </tr>
  </thead>
  <tbody>
    <tr className=\\"item-row\\">
      <td>
        Aged Brie
      </td>
      <td>
        3
      </td>
      <td>
        60
      </td>
    </tr>
    <tr className=\\"item-row\\">
      <td>
        Backstage passes to a TAFKAL80ETC concert
      </td>
      <td>
        8
      </td>
      <td>
        15
      </td>
    </tr>
    <tr className=\\"item-row\\">
      <td>
        Sulfuras, Hand of Ragnaros
      </td>
      <td>
        80
      </td>
      <td>
        1000
      </td>
    </tr>
    <tr className=\\"item-row\\">
      <td>
        Generic, very boring item
      </td>
      <td>
        2
      </td>
      <td>
        100
      </td>
    </tr>
    <tr className=\\"item-row\\">
      <td>
        Conjured Item
      </td>
      <td>
        40
      </td>
      <td>
        10
      </td>
    </tr>
  </tbody>
</ForwardRef>"
`;
